hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/misc/BesuNativeLibVerificationTest.java

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.services.bdd.suites.misc;
+
+import static com.hedera.services.bdd.junit.TestTags.LONG_RUNNING;
+import static com.hedera.services.bdd.spec.HapiSpec.hapiTest;
+import static com.hedera.services.bdd.spec.utilops.FakeNmt.restartNetwork;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.blockingOrder;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.doAdhoc;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.doingContextual;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.freezeOnly;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.sleepForSeconds;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.waitForAny;
+import static com.hedera.services.bdd.suites.HapiSuite.GENESIS;
+import static com.hedera.services.bdd.suites.regression.system.LifecycleTest.*;
+import static org.hiero.consensus.model.status.PlatformStatus.ACTIVE;
+import static org.hiero.consensus.model.status.PlatformStatus.STARTING_UP;
+
+import com.hedera.services.bdd.junit.HapiTest;
+import com.hedera.services.bdd.junit.hedera.NodeSelector;
+import com.hedera.services.bdd.suites.regression.system.LifecycleTest;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Tag;
+
+@Tag(LONG_RUNNING)
+// Order to be last as it will restart the network and halt if the lib is not present
+@Order(Integer.MAX_VALUE)
+public class BesuNativeLibVerificationTest implements LifecycleTest {
+
+    @HapiTest
+    public Stream<DynamicTest> besuNativeLibVerificationHaltsIfLibNotPresent() {
+
+        final var envOverrides = Map.of("contracts.evm.nativeLibVerification.halt.enabled", "true");
+
+        return hapiTest(blockingOrder(
```

Comment:
It looks like there is a call to disable for example the `Blake2bfMessageDigest` by calling a `disableNative` method.  Can this fact be used for comprehensive testing?